file2.js

```diff
@@ -0,0 +1,558 @@
+        postUnignore.files = [];
+
+        var ignoreUrl = $(this).data("ignore");
+        var postIgnore = {};
+        postIgnore.projectId = projectId;
+        var hasError = false;
+
+        utils.post(unignoreUrl, postUnignore, function() {
+
+          //prepare post cycle, we only want to show feedback on the last iteration
+          var checkError = function(e) {
+            if (e.error) {
+              hasError = true;
+            }
+          };
+
+          var onPostError = function() {
+            hasError = true;
+          };
+
+          var onLastIterationSuccess = function(e) {
+            if (e.error) {
+              $(".IgnoreFilesFeedback").showError(e.error);
+            } else {
+              $(".IgnoreFilesFeedback").showSuccess("Saved");
+              ProjectSettings.prototype.changesFooter(".welcome__alert");
+            }
+          };
+          var onLastIterationError = function(e) {
+            $(".IgnoreFilesFeedback").showError(e.error);
+          };
+
+          for (i = 0, j = selected.length; i < j; i += chunk) {
+
+            postIgnore.files = selected.slice(i, i + chunk);
+            var lastIteration = i + chunk > selected.length;
+
+            if (!lastIteration) {
+              utils.post(ignoreUrl, postIgnore, checkError, onPostError);
+            }
+            else {
+              utils.post(ignoreUrl, postIgnore, onLastIterationSuccess, onLastIterationError);
+            }
+
+          }
+        });
+      });
+    };
+
+    ProjectSettings.prototype.bindPublicProject = function() {
+
+      var modalProject = $("#visibilityProjectModal");
+
+      var bindPublicOrPrivate = function() {
+        var $btn = $("#ChangeProjectVisibility");
+        var feedback = $(".ProjectFeedback");
+
+        var postVariable = {};
+        postVariable.projectId = $btn.data("projectid");
+        postVariable.setAsPublic = $btn.data("ispublic");
+
+        var url = $btn.data("url");
+        utils.post(url, postVariable, function(e) {
+          if (e.error) {
+            modalProject.modal("hide");
+            feedback.showError("Failed to change project visibility.");
+          } else {
+            window.location.reload();
+          }
+        }, function() {
+          modalProject.modal("hide");
+          feedback.showError("Failed to change project visibility.");
+        });
+      };
+
+      $(".change-visibility").click(function(e) {
+        e.preventDefault();
+        bindPublicOrPrivate();
+      });
+
+    };
+
+    ProjectSettings.prototype.changesFooter = function(footer) {
+      $(footer).fadeIn("fast");
+    };
+
+    ProjectSettings.prototype.bindProjectAnalysis = function() {
+      $(".js_project_remote_analysis").on('click', function() {
+        var $this = $(this);
+        var url = $this.data('url');
+        var payload = {};
+        payload[$this.attr('name')] = $this.prop('checked');
+        utils.JSend(url, payload,
```

Comment:
![Codacy](https://app.dev.codacy.org/assets/images/favicon.png) Issue found: [A function with a name starting with an uppercase letter should only be used as a constructor. (new-cap)](https://app.dev.codacy.org/app/codacytestuser/CheckPrCommentsPublicProject/pullRequest?prid=7992)